sktime/forecasting/mapa.py

```diff
@@ -0,0 +1,506 @@
+                    "Weights must have the same length "
+                    "as the number of aggregation levels."
+                )
+            weights = np.array(self.weights).reshape(-1, 1)
+        else:
+            weights = np.ones((forecasts.shape[0], 1))
+
+        weights = weights / weights.sum()
+
+        if self.forecast_combine == "mean":
+            return np.mean(forecasts, axis=0)
+        elif self.forecast_combine == "median":
+            return np.median(forecasts, axis=0)
+        elif self.forecast_combine == "weighted_mean":
+            return np.average(forecasts, axis=0, weights=weights.ravel())
+        else:
+            raise ValueError(
+                f"Unsupported forecast combination method: {self.forecast_combine}"
+            )
+
+    def _update(self, y, X=None, update_params=True):
+        """Update with new data following MAPA methodology."""
+        y = check_series(y)
+
+        if isinstance(y, pd.Series):
+            y = pd.DataFrame(y)
+        if y.columns.empty:
+            y.columns = self._y_cols
+
+        y = self._ensure_positive_values(y)
+        y = self._handle_missing_data(y)
+
+        for level in self.aggregation_levels:
+            try:
+                y_agg = self._aggregate(y, level)
+                y_agg.columns = self._y_cols
+
+                if update_params:
+                    if hasattr(self.forecasters[level], "update"):
+                        self.forecasters[level].update(y_agg, update_params=True)
+                    else:
+                        self.forecasters[level].fit(y_agg)
+
+            except Exception as e:
+                warn(f"Failed to update level {level}: {str(e)}")
+                continue
+
+        return self
+
+    @classmethod
+    def get_test_params(cls, parameter_set="default"):
+        """Return testing parameter settings for the estimator."""
+        from sktime.forecasting.trend import PolynomialTrendForecaster, TrendForecaster
+
+        params = [
+            {
+                "aggregation_levels": [1, 2, 3],
+                "base_forecaster": NaiveForecaster(strategy="mean"),
+                "imputation_method": "ffill",
+                "decompose_type": "multiplicative",
+                "forecast_combine": "mean",
+            },
+            {
+                "aggregation_levels": [1, 4, 6],
+                "base_forecaster": TrendForecaster(regressor=None),
+                "imputation_method": "interpolate",
+                "decompose_type": "additive",
+                "forecast_combine": "median",
+            },
+            {
+                "aggregation_levels": [1, 3, 6],
+                "base_forecaster": PolynomialTrendForecaster(degree=2),
+                "imputation_method": "ffill",
+                "decompose_type": "multiplicative",
+                "forecast_combine": "weighted_mean",
+                "weights": [0.5, 0.3, 0.2],
+            },
+        ]
+
```

Comment:
we should also a parameter set that has as many defaults as possible.
In this case it would be the "all defaults" setting, `{}`.